packages/@aws-cdk/aws-cloudformation/lib/aws-sdk-js-caller/index.js

```diff
@@ -0,0 +1,52 @@
+const AWS = require('aws-sdk');
+
+exports.handler = async function(event, context) {
+  try {
+    console.log(JSON.stringify(event));
+
+    if (event.ResourceProperties[event.RequestType]) {
+      const { service, action, parameters } = event.ResourceProperties[event.RequestType];
+      const awsService = new AWS[service]();
+      await awsService[action](parameters).promise();
+    }
+
+    await respond('SUCCESS', 'OK');
```

Comment:
The physical resource ID is actually a pretty important thing to get right when defining custom resources. I am wondering if we can do better than logStreamName

Reply:
Indeed, if we want to avoid calling the `onDelete` during a `UPDATE_COMPLETE_CLEANUP_IN_PROGRESS` when a parameter of the `onUpdate` changes we need something like a constant here, no? can we use the `LogicalResourceId`?

Reply:
This is a clever idea.  To be useful, this resource is going to be an exercise in metaprogramming.

Logical ID wouldn't be good enough. I think a template string of sorts will need to come from the parameter. Also wonder if we might want to call multiple functions.

At least we'd want to explode the API result into a set of output attributes, I think.

Reply:
+1, I think we will need to let users specify the path to where the physical ID can be found in the response JSON (at least), and this mechanism can also be used to produced a bunch of "GetAtt" attributes.

Reply:
In the response only or also in the request? If you look at the example with the `putTargets` call what would you specify as physical ID?

Reply:
> In the response only or also in the request? If you look at the example with the `putTargets` call what would you specify as physical ID?

The physical resource ID parameters needs to be able to be JSONPath member. It means it's not an ID by itself, but it's instructions for how to get the ID out of the API call response. So it should be able to be something like:

```
$.CreateThingResponse.ThingArn
```

And the custom resource should get the value out of the response object. Probably should be able to be different for both Create and Update calls.

Now, whether it can be EITHER a literal or a JSONPath I'm not sure on. I'm sure there are cases in which a literal would make sense, so I'm not against disallowing it, but the JSONPath-style query needs to be possible.

Reply:
P.S: When I say JSONPath, I'm cool with faking it by splitting on `.` and leaving it there, I don't mean literally importing a jsonpath library, we don't have a good enough story around bundled lambda dependencies to make that work.